macros/sql/get_tables_by_pattern_sql.sql

```diff
@@ -11,6 +11,7 @@
             case table_type
                 when 'BASE TABLE' then 'table'
                 when 'EXTERNAL TABLE' then 'external'
+                when 'FOREIGN' then 'external'
```

Comment:
Of course I could add separate foreign type, which would require adding new relation type here:
```
class RelationType(StrEnum):
    Table = 'table'
    View = 'view'
    CTE = 'cte'
    MaterializedView = 'materializedview'
    External = 'external'
```
and/or add this case specifically to `postgres__get_tables_by_pattern_sql` instead of `default__get_tables_by_pattern_sql`, but I think that would be an overkill.

Reply:
I agree that making a separate foreign type would be overkill, but I don't hate the idea of getting a focused `postgres__get_tables_by_pattern_sql` version of the macro, or at least a subset. 

What about a new macro that generates the case statement for different table types? Something like 
```
{% macro map_table_types(table_type) %}
  {{ return adapter.dispatch(yada yada) }} 
{% endmacro  %}

{% macro default__map_table_types() %}
 case table_type
   when 'BASE TABLE' then 'table'
   ...

{% endmacro %}

{% macro postgres__map_table_types() %}
 case table_type
   when 'BASE TABLE' then 'table'
   ...
   when 'FOREIGN' then 'external'

{% endmacro %}
```

and then get_tables_by_pattern_sql swaps out the case statement for a call to `map_table_types()` (I don't like the macro name; feel free to pick something better). 
Handling an adapter-specific piece of functionality that other adapters either shouldn't know about or might have an adverse reaction to in the future (living in fear of Snowflake one day adding a FOREIGN type which is some other wild thing 😱) feels like _exactly_ the sort of thing that multiple dispatch lends itself to with minimal overhead. 

How does that sound to you?

Reply:
Sounds reasonable to me, will update this PR today.